src/routes/user/index.ts

```diff
@@ -250,17 +255,30 @@ router.put(
     const body = req.body as PutUserSubscriptionRequest;
     const params = req.params as PutUserSubscriptionRequestParams;
 
-    const user = await User.findById(
-      params.id,
-      projection(keysOf<PutUserSubscriptionResponse>())
-    );
+    const user = await User.findById(params.id);
     if (!user) {
       throw new NotFound(ErrorMessage.USER_NOT_FOUND);
     }
 
+    if (user.subscriptions.includes(body.subscription)) {
+      throw new BadRequest(ErrorMessage.SUBSCRIPTION_ALREADY_EXISTS);
+    }
+
+    const channel = await Channel.findById(body.subscription);
+    if (!channel) {
+      throw new NotFound(ErrorMessage.CHANNEL_NOT_FOUND);
+    }
+
     user.subscriptions.push(body.subscription);
     user.save();
 
+    if (!channel.subscribers) {
+      channel.subscribers = 1;
```

Comment:
We should set this to default 0 in the schema, this way we dont have to check if it's initialized

Reply:
Note, this will probably require a "migration", we dont have to write it now, but maybe adding an issue with a list of things to migrate before we do another release would be good